### PR TITLE
Review by HarveySouth as PR

### PR DESCRIPTION
**Pull request by HarveySouth**

### CODE REVIEW

File: 01_data_exploration.ipynb [hunk 1]
```diff
@@ -10,17 +10,17 @@
     "# 1. Data Exploration\n",
```
Comment (HarveySouth) [inline 01_data_exploration.ipynb:10]: <p>Could: Phrase the learning outcomes in Imperative sentences.</p><p> </p><p>The second outcome is incomplete&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (HarveySouth) [inline 01_data_exploration.ipynb:10]: <p>It might be nice to link to further reading on the Met Office Machine Learning best practices, or some other accessible resource around these important topics. Some could be found in this <a href="https://docs.google.com/document/d/1LuHU8KcvleH7RkRitQux5LtxfZkA2SRxfNnTt6AqSsY/" target="_blank">annotated reading resource produced by a recent workshop on reproducibility</a>, in particular I thought this resource on ML Pitfalls was very accessible and concise: <a href="https://arxiv.org/abs/2108.02497" target="_blank">How to avoid machine learning pitfalls: a guide for academic researchers, Michael A. Lones</a></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (HarveySouth) [inline 01_data_exploration.ipynb:10]: <p>Could: Have some small contextual elaboration to example these issues e.g. a classifier that predicts "no rotors" regardless of input will have an accuracy of 97.43% </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (HarveySouth) [inline 01_data_exploration.ipynb:10]: <p>This cell causes an exception</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

[the remark this answers is not in the document]
Comment (HarveySouth) [reply 01_data_exploration.ipynb:10]: Sorry, was hoping this would get collapsed

**Comment by HarveySouth** [comment]
Strange output from reviewNB 👎

**Comment by HarveySouth** [comment]
Could: send xbt_imeta_flowchart.png to the images directory and update the reference to the image in the notebook

File: 03_algorithm_selection.ipynb [hunk 2]
```diff
@@ -17,7 +17,7 @@
    "source": [
```
Comment (HarveySouth) [inline 03_algorithm_selection.ipynb:17]: <p>This cell throws error, seems to be an old cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (HarveySouth) [inline 03_algorithm_selection.ipynb:17]: <p>It might be nice to mention that while the classical definition for a perceptron is around activations, the term is used ambiguously in text e.g. a multi-layer perceptron can simply mean a feedforward network where the outputs of a node are the weights and biases without thresholding or activations (leading to an output range in the set of real numbers). The expansion of this explanation could also allow for a description of two common ML problems: exploding and vanishing gradient. Where output restrictions like activations reduce exploding gradient issues and typically architecture decisions remedy vanishing gradients (RNN variants, or Skip/residual connections come to mind) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

**Comment by HarveySouth** [comment]
Could: Indicate in the notebooks, which conda environment is supposed to be used

File: 04_improving_performance.ipynb [hunk 3]
```diff
@@ -36,7 +36,7 @@
     "* Understand basic approaches to improving training data through resampling approaches\n",
```
Comment (HarveySouth) [inline 04_improving_performance.ipynb:36]: <p>Could: extend hyperparameter tuning tutorial with a test set evaluation from the best model. It would be interesting to discuss, but my own idea of reporting a good estimate of model performance involves finding the best hyperparameters that maximize the desired metric on the validation set, and then applying exactly the same training process to the model, trained on the training and validation data aggregated, tested against the test set since step/epoch are hyperparameters the final model inference metric is reportable. It is of course possible to report just the whole training process (which should be done anyway) and leave interpretation to the user/reader, but not sure what a best practice in this case would look like.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: 01_data_exploration.ipynb (hunk 1 above)
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>I've added some additional explanatory text.</p>

[the remark this answers is not in the document]
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>done</p>

[the remark this answers is not in the document]
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>I'm not sure what you mean about using the frequency distribution for Y, could you elaborate?</p>

[the remark this answers is not in the document]
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>done</p><p> </p>

Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>at some point i changed variable names and for some reason this cell didn't get updated (not sure  how i ran the notebook previously). Fixed niow</p>

[the remark this answers is not in the document]
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>that's weird, not sure whats goinh on there. Might be an issue with your conda environment. Its working OK for me. We'll see whether there are any problems during the tutorial.</p>

[the remark this answers is not in the document]
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>Can resolve this easily by just spacing out the plots a bit more using a 2x2 layout rather than 1x4, which I've now done.</p>

**Comment by stevehadd** [comment]
<p>Filled in now.</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by stevehadd** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: 01_data_exploration.ipynb (hunk 1 above)
Comment (stevehadd) [reply 01_data_exploration.ipynb:10]: <p>In this case, I actually want them to use the documents that we (the met office) have produced. But I cant put the link into the document, since that's on sharepoint and not accesible generally. I've added a sentence reminding people to look at the document on sharepoint.</p>

**Comment by stevehadd** [comment]
<p>Added a link to the met office document, as stated in comment in previous notebook.</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by stevehadd** [comment]
<p>yes, thats rather vagyue, I've reworded it.</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: 01_data_exploration.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarveySouth) [reply 01_data_exploration.ipynb:10]: <p>To elaborate: in this case the classes are imbalanced, so the counts for no rotors read into the 2000s while the for rotors data the counts are in the 40s, I thought since a comparison is being made, using the "percentage of the data that is of this histogram bin" would help for comparison.</p><p> </p><p>Now that I look at it though, the plots scale anyway, so the plots would look exactly the same. </p>

**Comment by stevehadd** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by stevehadd** [comment]
<p>Completed.</p><p> </p><p> </p><p>	</p>

---

 <a href='https://app.reviewnb.com/informatics-lab/ml_weather_tutorial/pull/21/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: 03_algorithm_selection.ipynb (hunk 2 above)
Comment (stevehadd) [reply 03_algorithm_selection.ipynb]: <p>It seems to have gotten out of order, it needed to be moved down to after the X_test and y_test were defined.</p>

File: 04_improving_performance.ipynb (hunk 3 above)
Comment (stevehadd) [reply 04_improving_performance.ipynb]: <p>I think I should definitely add this. I want to merge the changes from this branch into main ahead of the tutorial, so I've but this suggestion in a separate issue #22.</p>
